### ISSUE TITLE
What's the status?

### ISSUE THREAD

**Issue by liamoc** [pierreyoda/micropolis-rs#14]
Hi, I'm the author of the [Micro Entertainment Pack](https://liamoc.itch.io/) a collection of games written in Rust using my tesserae tile graphics library. 

I was thinking it would be super cool to write a port of micropolis using an interface from my library, and I found that you seem to have already ported it to Rust! 

What's the status of this project? Would I be able to take the micropolis_core crate, add an interface and have the full simcity experience? Or is there still more work to be done?

**Comment by pierreyoda** [comment]
Hi liamoc, thank you for your interest !

I'd say the `micropolis_core` crate is about 60% complete, I'm actively working on it whenever I can.

I would definitely be exited by another interface, I'm personally working on one with React and Next.js.

**Comment by liamoc** [comment]
Great! I am starting to work on a frontend now. Perhaps a silly question: What's the top-level structure I should use for the entire game state? At first I thought it was perhaps City, but the only constructor for this creates a City with an entirely blank map of dirt. Then I saw the `MicropolisCoreInterfacer` structure but that requires a `MicropolisCoreOptions` structure which has non-public fields and no constructor. So I'm a bit unsure where to begin. My intuition says that I should be able to run a map generator and start a city on a random map, then extract the `TileMap` and start drawing, but I'm stumbling on the first part. Any suggestions?

**Comment by pierreyoda** [comment]
Yes the initialisation story is not quite there yet, I would definitely like to offer a builder pattern accessible from the WASM interface. I will create an issue to do just this.

**Comment by liamoc** [comment]
Great! I will follow progress closely.

**Comment by thangtq-pionero** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
foshougua, @Jeeyb, @KananRengaraju, @AustinGreen, @jtraglia, @janndriessen, @sourabhmarathe, @sorsby, @Shantanu-Singh, @marcellobardus, @nityanantan23, @vvinteresting, @dhelbegor, @godkingjay, @fracek, @mikker, @cosmix, @necmettindev, @eunseong-theori, @reubinleo
